frontend/src/components/pages/CampOverview/ViewCamperModal.tsx

```diff
@@ -0,0 +1,184 @@
+import React from "react";
+
+import {
+  Box,
+  ModalOverlay,
+  ModalContent,
+  Modal,
+  ModalFooter,
+  Button,
+  Text,
+  Table,
+  Tbody,
+  Tr,
+  Td,
+  TableContainer,
+  Textarea,
+} from "@chakra-ui/react";
+
+import { Camper } from "../../../types/CamperTypes";
+import { CamperDetailsBadgeGroup } from "./CampersTable/CamperDetailsBadge";
+import PickupDropoffTableRow from "./PickupDropoffTableRow";
+
+type ViewCamperModalProps = {
+  viewCamperModalIsOpen: boolean;
+  viewCamperOnClose: () => void;
+  camper: Camper;
+};
+
+const ViewCamperModal = ({
+  viewCamperModalIsOpen,
+  viewCamperOnClose,
+  camper,
+}: ViewCamperModalProps): JSX.Element => {
+  return (
+    <Modal
+      isOpen={viewCamperModalIsOpen}
+      onClose={viewCamperOnClose}
+      isCentered
+      preserveScrollBarGap
+    >
+      <ModalOverlay />
+      <ModalContent maxWidth="600px" maxHeight="75%">
+        <TableContainer ml={8} pr={8} overflowY="auto">
+          <Table variant="simple">
+            <Tbody>
+              <Tr>
+                <Td pl={0} pr={0} pt={8} borderBottom="2px">
+                  <Text textStyle="displayLarge">
+                    {camper.firstName} {camper.lastName}{" "}
+                    <Text as="span" fontWeight="400">
+                      {" "}
+                      | Age: {camper.age}
+                    </Text>
+                  </Text>
+                  <Text textStyle="displaySmallRegular" pb={2}>
+                    Amount Paid: $
+                    {Object.values(camper.charges).reduce((a, b) => a + b)}
+                  </Text>
+                  <CamperDetailsBadgeGroup camper={camper} paddingLeft="0px" />
+                </Td>
+              </Tr>
+
+              <Tr>
+                <Td pl={0} pr={8} borderBottom="2px">
+                  <Text textStyle="displaySmallBold" pb={2}>
+                    Earliest Drop-off and Latest Pick-up
+                  </Text>
+                  <TableContainer>
+                    <Table variant="unstyled">
+                      <Tbody>
+                        <PickupDropoffTableRow
```

Comment:
hmm I see that you need the date prop to be a number for the findTime function. I think you could still use the map function here but map over numbers instead of the actual days

Reply:
Done.